test/unit-test/cellular_at_core_utest.c

```diff
@@ -0,0 +1,932 @@
+/*
+ * cellular_library v0.0.1
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file cellular_at_core_utest.c
+ * @brief Unit tests for functions in cellular_at_core.h.
+ */
+
+#include <string.h>
+#include <stdbool.h>
+
+#include "unity.h"
+
+/* Include paths for public enums, structures, and macros. */
+#include "cellular_platform.h"
+
+#include "cellular_at_core.h"
+
+/**
+ * @brief Cellular sample prefix string input.
+ */
+#define CELLULAR_SAMPLE_PREFIX_STRING_INPUT               "+CPIN:READY"
+
+/**
+ * @brief Cellular sample prefix string wron input.
+ */
+#define CELLULAR_SAMPLE_PREFIX_STRING_WRONG_INPUT          "+CPINREADY"
+
+/**
+ * @brief Cellular sample prefix string large input.
+ */
+#define CELLULAR_SAMPLE_PREFIX_STRING_LARGE_INPUT          "+CPIN:Story for Littel Red Riding Hood: Once upon a time there was a dear little girl who was loved by every one who looked at her, but most of all by her grandmother, and there was nothing that she would not have given to the child. Once she gave her a little cap of red velvet, which suited her so well that she would never wear anything else. So she was always called Little Red Riding Hood."
+
+/**
+ * @brief Cellular sample prefix string output.
+ */
+#define CELLULAR_SAMPLE_PREFIX_STRING_OUTPUT               "READY"
+
+/**
+ * @brief Cellular sample string leading white space.
+ */
+#define CELLULAR_SAMPLE_STRING_LEADING_WHITE_SPACE         "    response"
+
+/**
+ * @brief Cellular sample string trailing white space.
+ */
+#define CELLULAR_SAMPLE_STRING_TRAILING_WHITE_SPACE        "response    "
+
+/**
+ * @brief Cellular sample string all white space.
+ */
+#define CELLULAR_SAMPLE_STRING_ALL_WHITE_SPACE             "    response    "

```

Comment:
`CELLULAR_SAMPLE_STRING_ALL_WHITE_SPACE --> CELLULAR_SAMPLE_STRING_LEADING_AND_TRAILING_WHITE_SPACE`?

Reply:
this sounds good, thanks.